### ISSUE TITLE
Add SSE-S3 encryption support for S3 inventory

### ISSUE THREAD

**Issue by danielwhite** [terraform-providers/terraform-provider-aws#5046]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

The AWS Golang SDK is unable to construct a valid request for creating or updating an S3 inventory configuration with SSE-S3 encryption enabled.

When fixed, the following should no longer occur when trying to create or import S3 inventory resources:

```
$ terraform apply
* aws_s3_bucket_inventory.sse_s3: Error putting S3 bucket inventory configuration: MalformedXML: The XML you provided was not well-formed or did not validate against our published schema

$ terraform import aws_s3_bucket_inventory.encrypt_sse_s3
* import aws_s3_bucket_inventory.encrypt_sse_s3 result: my-test-bucket:SSE-S3: aws_s3_bucket_inventory.encrypt_sse_s3: sse_s3 encryption is unsupported
```

The test suite currently ignores the case of SSE-S3 until the underlying issue is resolved:

```
make testacc TEST=./aws TESTARGS='-run=TestAccAWSS3BucketInventory'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -run=TestAccAWSS3BucketInventory -timeout 120m
[ ... ]
=== RUN   TestAccAWSS3BucketInventory_encryptWithSSES3
--- SKIP: TestAccAWSS3BucketInventory_encryptWithSSES3 (0.00s)
	resource_aws_s3_bucket_inventory_test.go:63: SSE-S3 is not supported by the SDK.
[ ... ]
```

### New or Affected Resource(s)

* aws_s3_bucket_inventory

### Potential Terraform Configuration

```hcl
resource "aws_s3_bucket" "bucket" {
  bucket = "my-test-bucket"
  acl    = "private"
}

resource "aws_s3_bucket_inventory" "sse_s3" {
  bucket = "${aws_s3_bucket.bucket.id}"
  name   = "SSE-S3"

  included_object_versions = "Current"

  schedule {
    frequency = "Daily"
  }

  destination {
    bucket {
      format = "CSV"
      bucket_arn = "${aws_s3_bucket.bucket.arn}"

      encryption {
        sse_s3 {}
      }
    }
  }
}
```

### References

* Original limited feature: https://github.com/terraform-providers/terraform-provider-aws/issues/532#issuecomment-401346040
* Underlying SDK issue: aws/aws-sdk-go#2015

**Comment by bflad** [comment]
Looks like this was fixed in v1.15.2 of the AWS Go SDK -- I removed the error handling, which will be released in version 1.37.0 of the AWS provider, likely middle of next week.

**Comment by bflad** [comment]
This has been released in [version 1.37.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1370-september-19-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
